aws-java-sdk-core/src/main/java/com/amazonaws/retry/PredefinedBackoffStrategies.java

```diff
@@ -43,6 +43,13 @@
     /** Maximum retry limit.  Avoids integer overflow issues. **/
     private static final int MAX_RETRIES = 30;
 
+    private static final long exponentialDelayHelper(int retriesAttempted, int baseDelay, int maxBackoffTime) {
+      long potentialWait = (1L << retriesAttempted) * baseDelay;
+      return (retriesAttempted > MAX_RETRIES) ? maxBackoffTime :
+              (potentialWait < 0  ? maxBackoffTime :
+              Math.min(potentialWait, maxBackoffTime));
+    }
+
```

Comment:
/Nitpick: do you mind if we push this method to the bottom of the class?

Also I suspect we can get away with just taking in "retriesAttempted" as a parameter since it's the `1 << retriesAttempted` line that we're trying to wrap. What if we did something like this:

```java
private static int calculateExponentialMultiplier(int retriesAttempted) {
    int multiplier = 1 << retriesAttempted;
    return multiplier >= 0 ? multiplier : Integer.MAX_VALUE;
}
```